boot/bootutil/include/bootutil/bootutil_public.h

```diff
@@ -266,6 +267,33 @@ int
 boot_read_swap_state(const struct flash_area *fa,
                      struct boot_swap_state *state);
 
+/**
+ * @brief Set next image application slot by flash area pointer
+ *
+ * @param fa pointer to flash_area representing image to set for next boot;
+ * @param active should be true if @fa points to currently running image
+ *        slot, false otherwise;
+ * @param confirm confirms image; when @p active is true, this is considered
+ *        true, regardless of passed value.
+ *
+ * It is users responsibility to identify whether @p fa provided as parameter
+ * is currently running/active image and provide proper value to @p active.
+ * Failing to do so may render device non-upgradeable.
+ *
+ * Note that in multi-image setup running/active application is the one
+ * that is currently being executed by any MCU core, from the pair of
+ * slots dedicated to that MCU core. As confirming application currently
+ * running on a given slot should be, preferably, done after functional
+ * tests prove application to function correctly, it may not be a good idea
+ * to cross-confirm running images.
+ * An application should only confirm slots designated to MCU core it is
```

Comment:
The above not is advice rather than requirement: that why it is not written it is forbidden, rather that it may not be good idea.
Cross confirming images may not be good idea, unless images can somehow communicate their state, otherwise you may end up confirming something that does not work.
If you assume one image to be leading and other following, and leading would be the only one to confirm, this may work, but if both applications hold the same capability user will end up in complicated logic to avoid things falling apart.

Reply:
I bet that in many cases only one executable has ability to confirm images and also there is the main application image which is able to verify whether dependent images are working correctly (centralized architecture).

Reply:
Yes, there are many cases like that. In such cases running application is supposed to figure out what slot is running on other MCU it is trying to confirm and set `active` flag accordingly. Of course the primary app is the taking responsibility for confirming something that does not work.
Problem with confirming something on other MCU, is that you do not even know whether this thing still works. Because confirm happens directly to flash, the app may be actually rebooting from test, because it fails and resets, when it gets OK flag from primary application and then gets into reboot loop.